imap/jmap_backup.c

```diff
@@ -1957,9 +1968,27 @@ static void restore_mailbox_cb(const char *mboxname, void *data, void *rock)
 
         message_set_from_mailbox(mailbox, msgno, msg);
```

Comment:
Right, the signature on this is:
```
EXPORTED void message_set_from_mailbox(struct mailbox *mailbox, unsigned int recno, message_t *m)
```

So it's a recno.

I think the best approach might just be to iterate the mailbox and check for next uid - otherwise we could use the finduid lookup for each in order to find the correct record.  There is a question whether we should try to improve the APIs here first, or hack around them :(  But it turns out this is all broken if it races with a cyr_expire repack!